v3/neckarsulm.py

```diff
@@ -0,0 +1,125 @@
+    has_fee: bool = ExtendedBooleanValidator()
+    opening_hours: str = StringValidator(max_length=255)
+    max_height: Decimal = DecimalValidator()
+
+
+class NeckarsulmConverter(CsvConverter):
+    neckarsulm_row_validator = DataclassValidator(NeckarsulmRowInput)
+
+    source_info = SourceInfo(
+        id='neckarsulm',
+        name='Stadt Neckarsulm',
+        public_url='https://www.neckarsulm.de',
+    )
+
+    header_mapping: dict[str, str] = {
+        'id': 'uid',
+        'name': 'name',
+        'kategorie': 'type',
+        'y-koord': 'lat',
+        'x-koord': 'lon',
+        'strasse': 'street',
+        'plz': 'postcode',
+        'stadt': 'city',
+        'anz_plaetze': 'capacity',
+        'anzcarsharing': 'capacity_carsharing',
+        'anzeladestation': 'capacity_charging',
+        'anzfrauenpark': 'capacity_woman',
+        'anzbehinderte': 'capacity_disabled',
+        'gebuehren': 'has_fee',
+        'open_time': 'opening_hours',
+        'maxhoehe': 'max_height',
+    }
+
+    type_mapping: dict[str, ParkingSiteTypeInput] = {
+        'Parkplatz': ParkingSiteTypeInput.OFF_STREET_PARKING_GROUND,
+        'Wanderparkplatz': ParkingSiteTypeInput.OFF_STREET_PARKING_GROUND,
+        'Parkhaus': ParkingSiteTypeInput.CAR_PARK,
+        'Tiefgarage': ParkingSiteTypeInput.UNDERGROUND,
+        # there is the value 'p+r', but as this is not a parking type, but an usecase, we just map this to off street parking ground
+        'p+r': ParkingSiteTypeInput.OFF_STREET_PARKING_GROUND,
+    }
+
+    def handle_csv(self, data: list[list]) -> ImportSourceResult:
+        import_source_result = self.generate_import_source_result(
+            static_parking_site_inputs=[],
+            static_parking_site_errors=[],
+        )
+
+        mapping: dict[str, int] = self.get_mapping_by_header(self.header_mapping, data[0])
+
+        # We start at row 2, as the first one is our header
+        for row in data[1:]:
+            input_dict: dict[str, str] = {}
+            for field in self.header_mapping.values():
+                input_dict[field] = row[mapping[field]]
+
+            try:
+                input_data: NeckarsulmRowInput = self.neckarsulm_row_validator.validate(input_dict)
+            except ValidationError as e:
+                import_source_result.static_parking_site_errors.append(
+                    ImportParkingSiteException(
+                        uid=input_dict.get('id'),
+                        message=f'validation error for {input_dict}: {e.to_dict()}',
+                    ),
+                )
+                continue
+
+            parking_site_input = StaticParkingSiteInput(
+                uid=str(input_data.uid),
+                name=input_data.name,
+                type=self.type_mapping.get(input_data.type),
+                lat=input_data.lat,
+                lon=input_data.lon,
+                address=f'{input_data.street}, {input_data.postcode} {input_data.city}',
```

Comment:
Are these mandatory / always provided? If not, the formatting should adapt to available info